r2dbc-spi-test/src/main/java/io/r2dbc/spi/test/Example.java

```diff
@@ -84,6 +89,87 @@ default void bindNull() {
             .verifyComplete();
     }
 
+    @Test
+    default void blobInsert() {
+        Mono.from(getConnectionFactory().create())
+            .flatMapMany(connection -> Flux.from(connection
+
+                .createStatement(String.format("INSERT INTO blob_test VALUES (%s)", getPlaceholder(0)))
+                .bind(getPlaceholder(0), Blob.from(Mono.just(ByteBuffer.wrap("abc".getBytes()))))
+                .execute())
+
+                .concatWith(close(connection)))
+            .as(StepVerifier::create)
+            .expectNextCount(1)
+            .verifyComplete();
+    }
+
+    @Test
+    default void blobSelect() {
+        blobInsert();
+
+        Mono.from(getConnectionFactory().create())
+            .flatMapMany(connection -> Flux.from(connection
+
+                .createStatement("SELECT * from blob_test")
+                .execute())
+                .flatMap(result -> result
+                    .map((row, rowMetadata) -> row.get("image", Blob.class)))
+                .flatMap(Blob::stream)
+                .map(ByteBuffer::array)
```

Comment:
Only read-write heap buffers are array-backed. Direct buffers will fail with `UnsupportedOperationException` and read-only buffers fail with `ReadOnlyBufferException`.